extensions/notebook/src/test/protocol/notebookUriHandler.test.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'mocha';
+import * as vscode from 'vscode';
+import * as sinon from 'sinon';
+
+import { NotebookUriHandler } from '../../protocol/notebookUriHandler';
+
+describe('Notebook URI Handler', function (): void {
+	let notebookUriHandler: NotebookUriHandler;
+	let showErrorMessageSpy: sinon.SinonSpy;
+	let executeCommandSpy: sinon.SinonSpy;
+
+	beforeEach(() => {
+		showErrorMessageSpy = sinon.spy(vscode.window, 'showErrorMessage');
+		executeCommandSpy = sinon.spy(vscode.commands, 'executeCommand');
+		notebookUriHandler = new NotebookUriHandler();
+	});
+
+	afterEach(function(): void {
+		sinon.restore();
+	});
+
+	it('should handle empty string null undefined gracefully', function (done): void {
+		notebookUriHandler.handleUri(vscode.Uri.parse(''));
+		sinon.assert.calledOnce(showErrorMessageSpy);
+
+		notebookUriHandler.handleUri(null);
+		sinon.assert.calledTwice(showErrorMessageSpy);
+
+		notebookUriHandler.handleUri(undefined);
+		sinon.assert.calledThrice(showErrorMessageSpy);
+
+		sinon.assert.neverCalledWith(executeCommandSpy, 'notebook.command.new');
+		done();
```

Comment:
Just get rid of the dones - those are only really necessary if you're doing some sort of callback/async stuff